edb/edgeql/compiler/conflicts.py

```diff
@@ -187,8 +187,18 @@ def _compile_conflict_select(
                 left=lhs, right=rhs,
             ))
 
-    insert_subject = qlast.Path(steps=[
-        s_utils.name_to_ast_ref(subject_typ.get_name(ctx.env.schema))])
+    # If the type we are looking at is BaseObject, then this must a
+    # conflict check we are synthesizing for an explicit .id. We need
+    # to ignore access policies in that case, since there is no
+    # trigger to back us up.
```

Comment:
I'd also comment that this is safe to do, because we prevent you from inserting into an abstract type.